### ISSUE TITLE
Add Centerline loss

### ISSUE THREAD

**Issue by valhassan** [NRCan/geo-deep-learning#502]
## Description
Connectivity is an important characteristic of tubular structures such as roads. A well-defined differentiable centerline loss combined with other losses such as dice loss potentially allows us to preserve this quality. Reference [paper](https://arxiv.org/abs/2003.07311) 

## Rationale
 GDL can extract tubular structures such as roads. A loss function capable of preserving connectedness will improve overall extraction quality. 

## Possible Implementation
- implement center line loss according to this [source code](https://github.com/jocpae/clDice)
- implement dice loss following this [source code](https://github.com/qubvel/segmentation_models.pytorch/blob/master/segmentation_models_pytorch/losses/dice.py)
- Combine both losses in a class "CLDice"
